### DISCUSSION TITLE
How to handle REST API @Version in Swagger and OpenAPI

### DISCUSSION THREAD

**Discussion by hittlle** [micronaut-projects/micronaut-core#7836] [Q&A]
Micronaut has a dedicated annotation @Version for developers to implement their REST API Versioning strategy, and with it, we can put the version number in request headers or in query parameters. That's great. I encountered a problem here, if I wanna put the version number in the request header, how should I configure OpenAPI/Swagger UI to reflect this? The thing here is that different versions of a API Endpoint may have distinct request and response payload structure. What should I do? If I cannot do that, I have to switch back to the URL based versioning strategy, thanks.

**Comment by chris-b7h** [comment]
Hello, Would anyone from Micronaut Framework be able to address this issue, please?

**Comment by altro3** [comment]
Versioning support has already been implemented and described in the documentation https://micronaut-projects.github.io/micronaut-openapi/latest/guide/index.html#versions

Also, support for different API versions is implemented using `@OpenAPIGroup` annotaion.

If you have questions, suggestions or problems, please create tickets here: https://github.com/micronaut-projects/micronaut-openapi/issues . Discussions are not reviewed regularly and rarely
